### PR TITLE
Update nodejs version running actions from 16 to 20.

### PR DESCRIPTION
**Pull request by DeinAlptraum**
Node16 has ben EOL'd September 2023 and support on Github Actions has been dropped in early 2024. As a result, all workflows requesting Node16 are already forced to run on Node 20 since 3rd of June 2024, see the announcement by Github [here](https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/).
This means the affected actions have been running on Node20 for two months already. Thus this change is purely cosmetic and should not break anything, but it should resolve warnings we are currently seeing on every CI job using this action:
`The following actions uses Node.js version which is deprecated and will be forced to run on node20: llvm/actions/install-ninja@main. For more info: https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/`

### CODE REVIEW

**Comment by DeinAlptraum** [comment]
Requesting review from @tstellar

**Review by asl — APPROVED** [review]
Overall, this LGTM

But this needs final approval from @tstellar :)

**Review by tstellar — APPROVED** [review]

**Comment by DeinAlptraum** [comment]
I don't have commit access, so one of you will have to merge

**Pull request merged**
